File: BasicCalculatorApp/app/src/main/java/com/example/basiccalculatorapp/MainActivity.java
package com.example.basiccalculatorapp;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

//    Objects
    private EditText no1;
    private EditText no2;
    private Spinner spinner;
    private TextView result;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

//        Initialize the objects
        no1 = findViewById(R.id.no1);
        no2 = findViewById(R.id.no2);
        spinner = findViewById(R.id.spinner);
        result = findViewById(R.id.resultTextView);
        Button button = findViewById(R.id.button);



//        Create an ArrayAdapter using the string array and a default spinner layout
        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(
                this,R.array.operations, android.R.layout.simple_spinner_item
        );

//         Specify the layout to use when the list of choices appears
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);

//        Apply the adapter to the spinner
        spinner.setAdapter(adapter);


        button.setOnClickListener(view -> {
            String data = spinner.getSelectedItem().toString();

            double n1 = Double.parseDouble(no1.getText().toString());
            double n2 = Double.parseDouble(no2.getText().toString());
            double res = 0d;

            switch(data)
            {
                case "+":
                   res = n1 + n2;
                    break;
                case "-":
                    res = n1 - n2;
                    break;
                case "x":
                    res = n1 * n2;
                    break;
                case "÷":
                    res = n1 / n2;
                    break;
                case "^":
                    res = Math.pow(n1,n2);
                    break;
                case "Choose":
                    Toast.makeText(this, "Choose an Operation.", Toast.LENGTH_SHORT).show();
                    break;
            }
            result.setText(String.valueOf(res));
        });
    }

}

File: ShayariApp/app/src/main/java/com/example/shayariapp/MainActivity.kt
package com.example.shayariapp

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.Gravity
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.shayariapp.databinding.ActivityMainBinding
import com.google.firebase.firestore.FirebaseFirestore
import kotlin.system.exitProcess

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding
    lateinit var db: FirebaseFirestore

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        db = FirebaseFirestore.getInstance()
        db.collection("Shayari").addSnapshotListener { value, error ->
            val list = arrayListOf<CategoryModel>()
            val data = value?.toObjects(CategoryModel::class.java)
            list.addAll(data!!)

            binding.recylerViewCategory.layoutManager = LinearLayoutManager(this)

            binding.recylerViewCategory.adapter = CategoryAdapter(this,list)
        }



        binding.btnMenu.setOnClickListener {
            checkMenuClick()
        }



        binding.navView.setNavigationItemSelectedListener {
            when (it.itemId) {
                R.id.share -> {
                    try {
                        val shareIntent = Intent(Intent.ACTION_SEND)
                        shareIntent.type = "text/plain"
                        shareIntent.putExtra(Intent.EXTRA_SUBJECT, "Hindi Shayari App")
                        var shareMessage = "\nLet me recommend you this application\n\n"
                        shareMessage =
                            """
                            ${shareMessage}https://play.google.com/store/apps/details?id=${BuildConfig.APPLICATION_ID}
                            
                            
                            """.trimIndent()
                        shareIntent.putExtra(Intent.EXTRA_TEXT, shareMessage)
                        startActivity(Intent.createChooser(shareIntent, "Choose one"))
                    } catch (e: Exception) {
                        //e.toString();
                    }
                    true
                }
                R.id.more -> {
                    val uri = Uri.parse("market://details?id=$packageName")
                    val myAppLinkToMarket = Intent(Intent.ACTION_VIEW, uri)
                    try {
                        startActivity(myAppLinkToMarket)
                    } catch (e: ActivityNotFoundException) {
                        Toast.makeText(this, " unable to find market app", Toast.LENGTH_LONG).show()
                    }
                    true
                }
                R.id.rate -> {
                    try {
                        startActivity(Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=$packageName")))
                    } catch (e: ActivityNotFoundException) {
                        startActivity(Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/details?id=$packageName")))
                    }
                    true
                }
                R.id.exit -> {
                    exitProcess(0)
                }
                else -> {
                    false
                }
            }
        }

    }

    private fun checkMenuClick() {
        if (binding.drawerLayout.isDrawerOpen(Gravity.LEFT)) {
            binding.drawerLayout.closeDrawer(Gravity.LEFT)
        } else {
            binding.drawerLayout.openDrawer(Gravity.LEFT)
        }
    }

    override fun onBackPressed() {
        checkMenuClick()
    }

}